### PR TITLE
connectorEnforcement field added along with modified test

### PR DESCRIPTION
**Pull request by sanghaniJ**
connector_enforcement field added to sql_database_instance resource
settings test case modified with new field 

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
cloudsql: added `connector_enforcement` field to `google_sql_database_instance` resource
```

### CODE REVIEW

**Comment by ScottSuarez** [comment]
Could you try pulling in the latest changes from main?

File: mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb [hunk 1]
```diff
@@ -1561,6 +1568,7 @@ func flattenSettings(settings *sqladmin.Settings) []map[string]interface{} {
 		"activation_policy":          settings.ActivationPolicy,
 		"availability_type":          settings.AvailabilityType,
 		"collation":                  settings.Collation,
+		"connector_enforcement":      settings.ConnectorEnforcement,
```
Comment (sanghaniJ) [inline mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb:1571]: Here, if we check the sqladmin.settings, it uses 0.92v while the field connectorEnforcement was added in 0.93. Hence the field is not taken into consideration and build is failing with the error connectorEnforcement field not found. Any thoughts on this?

**Comment by sanghaniJ** [comment]
@ScottSuarez  There are 2 tests failing for the resource but it doesn't seem like that is happening due to the field I've added. Those tests are failing in the current version as well. How do I proceed now?

File: mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb (hunk 1 above)
Comment (ScottSuarez) [reply mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb:1579]: I'm confused, what is the concern? If the sql version isn't a specific one the field may be omitted?

Comment (sanghaniJ) [reply mmv1/third_party/terraform/resources/resource_sql_database_instance.go.erb:1579]: That issue regarding older versions of sqladmin in use was fixed when it was upgraded from 0.92 to 0.97. But the current concerns are there are 2 tests failing for the resource but it doesn't seem like that is happening due to the field I've added. Even when run locally, Those tests are failing even when I use online build as well.

**Comment by ScottSuarez** [comment]
Hi Jay, would you be able to file a bug with the team that owns the api failing in the test. It's not clear to me why it's failing and claims to be transient. According to our reruns. This claim is not holding up

**Comment by sanghaniJ** [comment]
Actually I'm not authorised to do so, hence if possible can you do that on my behalf.

**Comment by melinath** [comment]
"transient error" should now be resolved /gcbrun

**Comment by ScottSuarez** [comment]
tracking failure with team b/254484541

**Comment by ScottSuarez** [comment]
Could you update the prefix length to 24 in the relevant faililng tests for google_compute_global_address. Not sure why, but that seems to be the recommendation given.

**Comment by sanghaniJ** [comment]
Sure, will do so and check if the cases pass

**Comment by sanghaniJ** [comment]
@ScottSuarez I tried updating the prefix length to 24 in the relevant failing tests still the error stays the same.

**Comment by ScottSuarez** [comment]
Lets `skip_vcr` on the following tests to unblock this change.

`TestAccSqlDatabaseInstance_SqlServerAuditConfig`
`TestAccSqlDatabaseInstance_withPrivateNetwork_withAllocatedIpRange`

These tests are failing in teamcity (our nightly runs). Could you also create a bug to track that we disabled them?
https://github.com/hashicorp/terraform-provider-google/issues/new?assignees=&labels=test+failure&template=10_test_failure.md&title=Failing+test%28s%29%3A+TestAccWhatever

**Comment by sanghaniJ** [comment]
@ScottSuarez I've added skipVcr and now the checks are passing. About the failing tests, I guess Stephen has created a [bug ticket](https://github.com/hashicorp/terraform-provider-google/issues/12902) for the failing 2 tests along with some other tests.

**Comment by sanghaniJ** [comment]
Also, there are several test cases with Failure due to the same issue, they were already added to SkipVcr.

**Comment by ScottSuarez** [comment]
Have you been able to run these tests locally and do they pass for you. 

I am curious if we can get a minimum config test passing. Without some sort of knowledge as to, yes I can confirm this works. I worry about merging. I could also perhaps try to run some tests locally.

**Comment by sanghaniJ** [comment]
I guess it has a similar issue for all such test cases, regarding the `Service Networking` which results into failure. I've tried running locally but the tests are failing.

**Comment by sanghaniJ** [comment]
Hi @ScottSuarez, Can you have a look. What I found was all the cases with Service Networking present are skipped as of now. Looks like it is blocking several cases. I guess it will be fine to proceed. Also wanted to ask,  did you get any findings running it locally?

**Comment by ScottSuarez** [comment]
Yeah, if we can't get any sort test to pass with this field, locally or otherwise. I'm against merging. If you can produce a configuration and deployment that works I am more comfortable checking this change in.

**Comment by sanghaniJ** [comment]
Just to be clear, the `serviceNetworking` field isn't added by me. It is already present and is the reason for the multiple test failures which are failing in TeamCity as well. The field I've added is `connectorEnforcement` and it is added in the test `testGoogleSqlDatabaseInstance_settings_secondary` which passes fine.

**Review by ScottSuarez — APPROVED** [review]
Ack, apologies here. It's been so long trying to sort this out the context was lost. Thanks for your patience here.

**Pull request merged**